### PR TITLE
Introduced 'py-libsonata' package

### PR DESCRIPTION
**Pull request by arsenius7**
* fetching source code from PyPI
 * removed 'python' variant for 'libsonata'

### CODE REVIEW

**Comment by arsenius7** [comment]
Another try with `py-libsonata`; this time using source tarball from PyPI as [recommended](https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#pypi-vs-github) in Spack docs.

This way we sacrifice link with Spack-deployed `libsonata` C++ library in favor of the link between `pip` and `Spack` view on `libsonata` Python package; which, imho, is more important.

Alternative proposal: again, a separate `py-libsonata` recipe, where the details common with `libsonata` recipe are moved into mixin class (I have not managed to get it working though :) ):
https://github.com/arsenius7/spack/commit/d5f19cf19408830937bd3a45b56c5f3d432d1e04

In general, it would be good if we can align on how to deploy BBP-developed hybrid C++ / Python libraries.
At the moment, it's a bit of a zoo, with at least two approaches for Python bindings:

1) Full CMake, where Python library is installed with `make install` (`mvdtool`, `brion`, `libsonata` prior to recent changes from @wizmer)

2) CMake triggered from `pip` (`py-basalt`, `[py-]morphio`, `[py-]libsonata`)

and different approaches for Spack recipe for option (2):

2a) recipe duplication `[py-]morphio`
2b) falling back to `make install` (`libsonata+python`)
2c) PyPI source tarball, as proposed here
2d) no C++ recipe for `py-basalt` (I guess there was no need so far)

**Comment by arsenius7** [comment]
retest please?

**Comment by wizmer** [comment]
1) I can only agree with myself by saying that having the ability to install a python package using pip is important.
2) I like this new tarball approach. And since PyPi only authorizes to upload a tarball once, it guaranties stability and unity. I don't know if we can still somehow make this work with the variants. Maybe passing them through as pip arguments @pramodk do you know ?

File: var/spack/repos/builtin/packages/py-libsonata/package.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+
+class PyLibsonata(PythonPackage):
+    """SONATA files reader"""
+
+    homepage = "https://github.com/BlueBrain/libsonata"
+    url = "https://pypi.io/packages/source/l/libsonata/libsonata-0.0.2.tar.gz"
+
+    version('0.0.2', sha256='ea6655d8f1cb79262b5c5ad613386e88635ff7a4745cf7ef5643fd3c59be9a55', preferred=True)
+
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-libsonata/package.py:16]: just curious : pypi tarball comes with pybind, highfive / all submodules?

Comment (arsenius7) [reply var/spack/repos/builtin/packages/py-libsonata/package.py:16]: Yes, header-only dependencies `pybind11`, `highfive`, `fmt` are bundled into the tarball (only files necessary for compilation + license, for a total of +200K).
`hdf5` is expected to be available in the system.

**Review by wizmer — APPROVED** [review]

**Pull request merged**